tft.py

```diff
@@ -0,0 +1,189 @@
+import json
+import glob
+import mmap
+import copy
+import re
+import os
+from json import JSONEncoder
+from cdragontoolbox.binfile import BinFile, BinHashBase, BinHashValue
+
+
+class NaiveJsonEncoder(JSONEncoder):
+    def default(self, other):  # pylint: disable=method-hidden
+        if isinstance(other, BinHashBase):
+            if other.s is None:
+                return other.hex()
+            return other.s
+        return other.__dict__
+
+
+class TFTTransformer:
+    def parse(self, input):
+        map22_file = os.path.join(input, "data", "maps", "shipping", "map22", "map22.bin")
+        character_folder = os.path.join(input, "data", "characters")
+
+        map22 = BinFile(map22_file)
+        items = self.parse_items(map22)
+        traits = self.parse_traits(map22)
+        champs = self.parse_champs(map22, traits, character_folder)
+
+        # clean up data for JSON export
+        [x.pop("internal") for x in traits]
+        [x.pop("internal") for x in items]
+
+        self.data = {"items": items, "traits": traits, "champs": champs}
+
+    def export(self, output, font_files=[], output_template=False):
+        os.makedirs(output, exist_ok=True)
+        template = json.dumps(self.data, cls=NaiveJsonEncoder, indent=4, sort_keys=True)
+
+        if output_template:
+            with open(os.path.join(output, "template.json"), "w", encoding="utf-8") as f:
+                f.write(template)
+
+        for language in font_files:
+            use = template
+            code = os.path.basename(language)[11:-4]
+
+            with open(language, "r", encoding="utf-8") as f:
```

Comment:
`"r"` is the default so you don't need to pass it explicitely (and it's usely omitted).